modular/uploader/upload_task.go

```diff
@@ -327,21 +337,25 @@ func (u *UploadModular) HandleResumableUploadObjectTask(ctx context.Context, tas
 					return ErrPieceStoreWithDetail(fmt.Sprintf("failed to put segment piece to piece store, piece_key: %s, error: %s",
 						pieceKey, err.Error()))
 				}
-				err = u.updatePieceCheckSum(task, data, isUpdate)
+				err = u.updatePieceCheckSum(task, data, isUpdate, uint64(readN))
 				if err != nil {
 					log.CtxErrorw(ctx, "failed to append integrity checksum to db", "error", err)
 					return ErrGfSpDBWithDetail("failed to append integrity checksum to db, error: " + err.Error())
 				}
 			}
 			if task.GetCompleted() {
 				var pieceChecksumList [][]byte
-				pieceChecksumList, err = u.getPieceCheckSumList(task, isUpdate)
+				pieceChecksumList, pieceSize, err = u.getPieceCheckSumListAndPieceSize(task, isUpdate)
 				if err != nil {
 					log.CtxErrorw(ctx, "failed to get object integrity hash", "error", err)
 					return err
 				}
+				if task.GetIsAgentUpload() && pieceSize != task.GetObjectInfo().GetPayloadSize() {
+					log.CtxErrorw(ctx, "payload size error", "db size", pieceSize, "objectInfo size", task.GetObjectInfo().GetPayloadSize())
```

Comment:
add object id to the err log, and `db size` is not descriptive. Can you make them like "expected" and "actual"